### DISCUSSION TITLE
next/image rendering using tailwind object-cover

### DISCUSSION THREAD

**Discussion by marcofranssen** [vercel/next.js#20942] [Help] [answered]
Currently I'm rendering my images using `tailwind` `object-cover` class. This allows me to use same sizes for images and crop those images automatically in my layout.

```jsx
export default function CoverImage({ title, src, slug, size }) {
  const image = (
    <img
      src={src}
      alt={`Cover Image for ${title}`}
      className={cn("object-cover", "w-full", {
        "h-36 md:h-60 lg:h-72": size != "large",
      })}
    />
  );

  return (
    <div
      className={cn("shadow-small", "p-5", "m-2", {
        "hover:shadow-medium transition-shadow duration-200": slug,
      })}
    >
      {slug ? (
        <Link href={`/blog/${slug}`}>
          <a aria-label={title}>{image}</a>
        </Link>
      ) : (
        image
      )}
    </div>
  );
}
```

See https://nextjs.marcofranssen.nl for an example.

Now I'm trying to utilize next/image in this component, but I struggle to get the same rendering behavior using `object-cover`.

This is my latest attempt.

```jsx
export default function CoverImage({ title, src, slug, size, dimensions }) {
  const image = (
    <Image
      width={dimensions.width}
      height={dimensions.height}
      src={src}
      alt={`Cover Image for ${title}`}
      className={cn("object-cover", "w-full", {
        "h-36 md:h-60 lg:h-72": size != "large",
      })}
    />
  );

  return (
    <div
      className={cn("shadow-small", "p-5", "m-2", {
        "hover:shadow-medium transition-shadow duration-200": slug,
      })}
    >
      {slug ? (
        <Link href={`/blog/${slug}`}>
          <a aria-label={title}>{image}</a>
        </Link>
      ) : (
        image
      )}
    </div>
  );
}
```

See here for the result. https://nextjs-blog-git-cover-image.marcofranssen.vercel.app/

Anybody here able to help me out?

**Comment by jamesmosier** [comment, marked as answer]
Have you tried adding the [`layout="fill"`](https://nextjs.org/docs/api-reference/next/image#layout) prop to the `Image` component and also removing the `width` and `height` props?

**Reply by marcofranssen** [reply]
Yes I tried. It seems there are some styles and divs injected arround the image that probably cause the tailwind classes to be overwritten or ignored. 

Maybe this is also a shortcoming of the next/image component at this point in time.

Just hoping someone with more in depth knowledge of the component knows how to do it or maybe contribute a improvement to the component to make this kind of usecases possible.

@timneutkens maybe you could shed some light here.

**Reply by jamesmosier** [reply]
I use `object-cover` with `next/image` successfully. Here is how I used it in one instance:

```js
<div className="relative" style={{ height: '600px' }}>
  <Image
    src="/my-image.jpg"
    alt="My Image"
    layout="fill"
    className="object-cover object-right"
  />
</div>
```

**Reply by marcofranssen** [reply]
@jamesmosier As I know my image dimensions ahead of time, do you think there would also be a way to achieve this using `width` and `height`?
